### PR TITLE
upgrade libdev experience; upgrade macos chassis dev experience; fix …

### PR DESCRIPTION
**Pull request by zackbrown**
…sticky cargo cache with include_dir for libdev, include app icons for macos

### CODE REVIEW

**Comment by zackbrown** [comment]
^ @warfaj this should be a big libdev experience upgrade; descriptions in commit message.  

Now `pax build` and `pax run` for macos will build for the current machine's active arch only via xcodebuild, which should 1. shorten build times, and 2. fix the spurious error message and silent half-failure we were sometimes getting from xcodebuild.  Note I have only tested this on arm64 — would be great to make sure you can still build for macos on your x86_64 machine.

Also, when in the new `libdev mode`, the compiler now copies directly from file system for `../pax-chassis-web` and `-macos`, instead of relying on `include_dir` (`include_dir` is still used for packaged userland builds of the CLI and compiler, for example if you install `pax-cli` from crates.io).  This fixes the cache stickiness problem that previously required both 1. cargo clean, and 2. rm -rf ......./chassis/...... — now you can freely make changes to the chassis template dir (@/pax-chassis-macos, for example) and immediately propagate them into codegenned builds with a simple `pax run --target macos` or from within clion with the Play button

Finally, there are now Pax icon assets built into the xcode dev harness template — I also bumped the target macOS sdk version along the way, which seemed to have introduced a "feature" where there's a "prohibited" 🚫 icon overlaying the app icon.  From what I've read, this is due to the fact that the executable is not code-signed and is expected.

File: pax-compiler/src/lib.rs [hunk 1]
```diff
@@ -671,7 +687,14 @@ fn clone_target_chassis_to_dot_pax(relative_chassis_specific_target_dir: &PathBu
     match RunTarget::from(target_str) {
         RunTarget::MacOS => {
 
-            persistent_extract(&CHASSIS_MACOS_DIR, &chassis_specific_dir).unwrap();
+            if libdevmode {
```
Comment (warfaj) [inline pax-compiler/src/lib.rs:690]: Pumped for this! no more cargo clean

**Comment by warfaj** [comment]
LGTM. Ran it locally and it builds/runs but I did notice that the logo had some weird artifact on top (https://drive.google.com/file/d/1YmimaCE9OULtjT6Bl95COfMIlHbH1Hr0/view?usp=share_link). After doing some research, it seems like that slash could mean it's not building the correct version. 
howtogeek.com/686351/why-are-some-mac-app-icons-crossed-out/

**Comment by warfaj** [comment]
Looked into it further. Ran a utility to figure out the build architecture of the app and it's right.
https://drive.google.com/file/d/134qk_EIGzpRHiAJkPnyxQCaOAQiCUqGD/view?usp=share_link
The artifact must be some sort of apple developer thing for custom icons. 
LGTM

**Review by warfaj — APPROVED** [review]

**Pull request merged**